File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int result = 0;
        
        int end = 0;
        
        int reach = 0;
        
        for(int i = 0; i < nums.size() - 1; i++) {
            reach = max(reach, i + nums[i]);
            
            if(reach >= nums.size()-1) {
                result++;
                break;
            }
            
            if(i == end) {
                result++;
                end = reach;
            }
        }
        
        return result;
        
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        result = 0
        reach = 0
        end = 0
        
        for i in range(len(nums) - 1):
            reach = max(reach, i+nums[i])
            if reach >= len(nums)-1:
                result += 1
                break
            
            if i == end:
                result += 1
                end = reach
        
        return result